hoodie-client/src/main/java/com/uber/hoodie/index/bloom/HoodieBloomIndex.java

```diff
@@ -300,22 +302,22 @@ public boolean isImplicitWithStorage() {
    * recordKey ranges in the index info.
    */
   @VisibleForTesting
-  JavaPairRDD<String, Tuple2<String, HoodieKey>> explodeRecordRDDWithFileComparisons(
+  JavaPairRDD<Tuple2<String, String>, Tuple2<String, HoodieKey>> explodeRecordRDDWithFileComparisons(
```

Comment:
@vinothchandar Just to be clear, you're suggesting this signature become  `JavaPairRDD<Tuple2<String, HoodieKey>>` and we rely on the List sorting to sort first based on `matchingFile` and then `HoodieKey`, correct?

Reply:
yes.. what do you think? we will save shuffling two more strings.

Reply:
Seems like it could work, I'll give it a shot. I mean, fewer things to shuffle is strictly better right?

Reply:
yes correct!